recipes/darknet/all/conanfile.py

```diff
@@ -0,0 +1,100 @@
+import os
+from conans import ConanFile, tools, AutoToolsBuildEnvironment
+from conans.errors import ConanInvalidConfiguration
+
+
+class DarknetConan(ConanFile):
+    name = "darknet"
+    license = "YOLO"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://pjreddie.com/darknet/"
+    description = "Darknet is a neural network frameworks written in C"
+    topics = ("darknet", "neural network", "deep learning")
+    settings = "os", "compiler", "build_type", "arch", "arch_build"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "OpenCV": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "OpenCV": False,
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _commit(self):
+        url = self.conan_data["sources"][self.version]["url"]
+        return url[url.rfind("/")+1:url.rfind(".")]
+
+    @property
+    def _lib_to_compile(self):
+        if not self.options.shared:
+            return "$(ALIB)"
+        else:
+            return "$(SLIB)"
+
+    @property
+    def _shared_lib_extension(self):
+        if self.settings.os == "Macos":
+            return ".dylib"
+        else:
+            return ".so"
+
+    def _activate_makefile(self, option):
+        makefileName = option.upper()
+        with tools.chdir(self._source_subfolder):
+            if getattr(self.options, option) == True:
+                tools.replace_in_file('Makefile', makefileName + '=0', makefileName + '=1')
+
+    def configure(self):
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("This library is not compatible with Windows")
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def requirements(self):
+        if self.options.OpenCV:
+            self.requires("opencv/3.4.3")
```

Comment:
there is no such package (yet)

Reply:
@SSE4 do you have some idea about moving OpenCV to CCI ?

Reply:
I suppose it's okay to submit an inclusion request, however:
1. it has lots of dependencies, which are not in CCI yet (e.g. ffmpeg, GStreamer, eigen, openblas, hurfbuzz, etc.)
2. it has system requirements, which are problematic. probably, it will be the first case, in which we will have to make decision how to handle it

Reply:
maybe, the first step might be the reduced version, which has only dependencies available on CCI right now and no system requirements. then we can add new things on demand.

Reply:
Good idea!

Reply:
please @SSE4 go ahead with the reduced version of opencv or its dependencies

Reply:
@SSE4 any news on OpenCV ?

Reply:
yes https://github.com/conan-io/conan-center-index/pull/2743